src/main/java/org/prebid/server/bidder/rubicon/RubiconBidder.java

```diff
@@ -1264,12 +1277,48 @@ private static Source makeSource(Source source, String pchain) {
                 .collect(Collectors.toList());
     }
 
+    private RubiconSeatBid updateSeatBids(RubiconSeatBid seatBid) {
+        final String buyer = seatBid.getBuyer();
+        final int networkId = NumberUtils.toInt(buyer, 0);
+        if (networkId <= 0) {
+            return seatBid;
+        }
+        final List<Bid> updatedBids = seatBid.getBid().stream()
+                .map(bid -> insertNetworkIdToMeta(networkId, bid))
+                .collect(Collectors.toList());
+        return seatBid.toBuilder().bid(updatedBids).build();
+    }
+
+    private Bid insertNetworkIdToMeta(int networkId, Bid bid) {
+        final ObjectNode bidExt = bid.getExt();
+        final ExtPrebid<ExtBidPrebid, ObjectNode> extPrebid = getExtPrebid(bid.getExt());
+        final ExtBidPrebid extBidPrebid = extPrebid != null ? extPrebid.getPrebid() : null;
+        final ExtBidPrebidMeta meta = extBidPrebid != null ? extBidPrebid.getMeta() : null;
+
+        final ExtBidPrebidMeta updatedMeta = meta != null
+                ? meta.toBuilder().networkId(networkId).build()
+                : ExtBidPrebidMeta.builder().networkId(networkId).build();
+
+        final ExtBidPrebid modifiedExtBidPrebid = extBidPrebid != null
+                ? extBidPrebid.toBuilder().meta(updatedMeta).build()
+                : ExtBidPrebid.builder().meta(updatedMeta).build();
+
+        final ObjectNode updatedBidExt = bidExt != null ? bidExt : mapper.mapper().createObjectNode();
+        updatedBidExt.set(PREBID_EXT, mapper.mapper().valueToTree(modifiedExtBidPrebid));
+
+        return bid.toBuilder().ext(updatedBidExt).build();
+    }
+
+    private ExtPrebid<ExtBidPrebid, ObjectNode> getExtPrebid(ObjectNode bidExt) {
+        return bidExt != null ? mapper.mapper().convertValue(bidExt, EXT_PREBID_TYPE_REFERENCE) : null;
```

Comment:
The `convertValue(..)` throws `IllegalArgumentException`. Should we care of it anywhere?